blockchain/tx_pool.go

```diff
@@ -899,7 +912,7 @@ func (pool *TxPool) add(tx *types.Transaction, local bool) (bool, error) {
 			return false, ErrUnderpriced
 		}
 		// New transaction is better than our worse ones, make room for it
-		drop := pool.priced.Discard(pool.all.Count()-int(pool.config.ExecSlotsAll+pool.config.NonExecSlotsAll-1), pool.locals)
+		drop := pool.priced.Discard(pool.all.Slots()-int(pool.config.ExecSlotsAll+pool.config.NonExecSlotsAll)+numSlots(tx), pool.locals)
```

Comment:
Arithmetic reordering seems to make it readable linearly in my view, but maybe not. 
```
drop := pool.priced.Discard(pool.all.Slots()+numSlots(tx)  -int(pool.config.ExecSlotsAll+pool.config.NonExecSlotsAll), pool.locals)
```

Also, a shortcut for long wording `pool.config.ExecSlotsAll+pool.config.NonExecSlotsAll` can be used later.

Reply:
I'll apply it later, thanks!